### PR TITLE
Fix BFunctionType toString method

### LINKED ISSUE #35356: ToString() does not show union types when they are as function arguments
**Description:**
```bal
    function (string? a) returns string e0 = (b) => "someString";
    io:println(e0);
```

Prints `function isolated function () returns (string)` where it should print `function isolated function (`**string?**`) returns (string)`

### PR DESCRIPTION
**Pull request by Nadeeshan96**
$title
Fixes #35356

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BFunctionType.java [hunk 1]
```diff
@@ -169,7 +169,8 @@ public String toString() {
         if (SymbolFlags.isFlagOn(this.flags, SymbolFlags.ANY_FUNCTION)) {
             stringRep = "function";
         } else {
-            stringRep = "function (" + (parameters != null ? getTypeListAsString(parameters) : "") + ")" +
+            stringRep = "function (" + (parameters != null ? getTypeListAsString(parameters) : "") +
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BFunctionType.java:172]: Shall we use `StringBuilder` in this method to improve the readability?

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BFunctionType.java [hunk 2]
```diff
@@ -169,7 +169,8 @@ public String toString() {
...
+                    (restType != null ? "," + ((BArrayType) restType).getElementType().toString() + "..." : "") + ")" +
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BFunctionType.java:173]: Shall we check that casting the `restType` into `BArrayType` will not cause a CCE?
If we always have the `restType` as an `ArrayType`, we can change the type of that field.

Comment (Nadeeshan96) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BFunctionType.java]: I checked this by building the ballerina-lang with tests and found out restType will be either `null` or instanceOf `BArrayType`.

**Review by HindujaB — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]

**Comment by Nadeeshan96** [comment]
Fix sent to master branch https://github.com/ballerina-platform/ballerina-lang/pull/35778. Hence closing this.

**Pull request closed without merging**